tests/admin_changelist/tests.py

```diff
@@ -429,6 +431,61 @@ def test_pk_in_search_fields(self):
         cl = ChangeList(request, Concert, *get_changelist_args(m))
         self.assertEqual(cl.queryset.count(), 0)
 
+    def test_custom_lookup_in_search_fields(self):
+        band = Group.objects.create(name='The Hype')
+        Concert.objects.create(name='Woodstock', group=band)
+
+        m = ConcertAdmin(Concert, custom_site)
+        m.search_fields = ['group__name__cc']
+
+        try:
+            Field.register_lookup(Contains, 'cc')
+            request = self.factory.get('/concert/', data={SEARCH_VAR: 'Hype'})
+            cl = ChangeList(request, Concert, *get_changelist_args(m))
+            self.assertEqual(cl.queryset.count(), 1)
+
+            request = self.factory.get('/concert/', data={SEARCH_VAR: 'Woodstock'})
+            cl = ChangeList(request, Concert, *get_changelist_args(m))
+            self.assertEqual(cl.queryset.count(), 0)
+        finally:
+            Field._unregister_lookup(Contains, 'cc')
+
+    def test_exclamation_mark_lookup_in_search_fields(self):
+        band = Group.objects.create(name='The Hype')
+        Concert.objects.create(name='Woodstock', group=band)
+
+        m = ConcertAdmin(Concert, custom_site)
+        m.search_fields = ['!name']
+
+        request = self.factory.get('/concert/', data={SEARCH_VAR: 'Woodstock'})
+        cl = ChangeList(request, Concert, *get_changelist_args(m))
+        self.assertEqual(cl.queryset.count(), 1)
+
+        request = self.factory.get('/concert/', data={SEARCH_VAR: 'woodstock'})
+        cl = ChangeList(request, Concert, *get_changelist_args(m))
+        self.assertEqual(cl.queryset.count(), 0)
+
+    def test_custom_lookup_transform_in_search_fields(self):
+        Band.objects.create(name='The Hype', nr_of_members=10)
+
+        m = BandAdmin(Band, custom_site)
+        m.search_fields = [
+            ('nr_of_members__cgt', int)
```

Comment:
I'm not sure if there are lookups where it's not the case, but comparisons such as `Choice.objects.filter(votes__gte='2')` seem to work fine with the value as a string so the "transform" stuff seems unnecessary, at this for the first version of this.